src/routes/getSkipSegmentsByHash.js

```diff
@@ -0,0 +1,69 @@
+const config = require('../config.js');
+const { db, privateDB } = require('../databases/databases.js');
+
+const getHash = require('../utils/getHash.js');
+const getIP = require('../utils/getIP.js');
+
+/**
+ * @typedef {Object} Segment
+ * @property {string} videoID YouTube video ID the segment is meant for
+ * @property {number[]} segment Tuple of start and end times in seconds
+ * @property {string} category Category of content to skip
+ * @property {string} UUID Unique identifier for the specific segment
+ */
+
+/**
+ * @param {string} prefix Lowercased hexadecimal hash prefix
+ * @param {string} hashedIP Custom hash of the visitor’s IP address
+ * @returns {Segment[]}
+ */
+function getSkipSegmentsByHash(prefix, hashedIP) {
```

Comment:
Hmm, I'm not sure about this. Is `category` really a large privacy concern?

Reply:
No. I just didn’t know if it even made sense to split by category at all? Maybe it does and I should add that as a choice to the URL.

Reply:
Segments from different categories may overlap, and would end up being filtered out by the `getVoteOrganisedSponsorTimes` logic. So, it has to be split by category.